source/Calamari.Common/Features/StructuredVariables/YamlFormatVariableReplacer.cs

```diff
@@ -25,7 +25,9 @@ public bool TryModifyFile(string filePath, IVariables variables)
             var variablesByKey = variables
                                  .Where(v => !octopusReservedVariablePattern.IsMatch(v.Key))
                                  .DistinctBy(v => v.Key)
-                                 .ToDictionary(v => v.Key, v => v.Value, StringComparer.OrdinalIgnoreCase);
+                                 .ToDictionary<KeyValuePair<string, string>, string, Func<string>>(v => v.Key,
+                                                                                                   v => () => variables.Get(v.Key),
```

Comment:
This can just be `v => variables.Get(v.Key)` I think.

Reply:
Because variable expansion can be recursive, and that structure contains every variable in the project (whether they will apply to this document or not), I was hesitant to eagerly expand all of them. Do you think that's a premature optimization?

Reply:
Ah, right, nah all good 👍